bundles/org.openhab.ui/web/src/pages/settings/rules/script/blockly-editor.vue

```diff
@@ -328,156 +266,388 @@
+            </value>
+          </block>
+          <block type="oh_say">
+            <value name="textToSay">
+              <shadow type="text">
+                <field name="TEXT">text to say</field>
+              </shadow>
+            </value>
+            <value name="deviceSink">
+              <shadow type="oh_audiosink_dropdown" />
+            </value>
+            <value name="voice">
+              <shadow type="oh_voices_dropdown" />
+            </value>
+          </block>
+        </category>
+        <category name="Logging/Output">
+          <block type="oh_log">
+            <value name="message">
+              <shadow type="text">
+                <field name="TEXT">abc</field>
+              </shadow>
+            </value>
+          </block>
+          <block type="oh_print">
+            <value name="message">
+              <shadow type="text">
+                <field name="TEXT">abc</field>
+              </shadow>
+            </value>
+          </block>
+        </category>
       </category>
-
       <sep />
-      <category name="Variables" colour="%{BKY_VARIABLES_HUE}" custom="VARIABLE" />
-      <category name="Functions" colour="%{BKY_PROCEDURES_HUE}" custom="PROCEDURE" />
+      <category
+        name="Variables"
+        colour="%{BKY_VARIABLES_HUE}"
+        custom="VARIABLE" />
+      <category
+        name="Functions"
+        colour="%{BKY_PROCEDURES_HUE}"
+        custom="PROCEDURE" />
     </xml>
   </div>
 </template>
 
 <style lang="stylus">
-.blockly-editor
-  display block
-  top calc(100%)
-  height calc(100%)
-  width 100%
-  .blockly
-    height 100%
-    .blocklyMainBackground
-      stroke inherit
-.blocklyDropDownDiv
-  z-index 9000
+.blockly-editor {
+  display: block;
+  top: calc(100%);
+  height: calc(100%);
+  width: 100%;
+
+  .blockly {
+    height: 100%;
+
+    .blocklyMainBackground {
+      stroke: inherit;
+    }
+  }
+}
+
+.blocklyDropDownDiv {
+  z-index: 9000;
+}
 </style>
 
 <script>
 import Blockly from 'blockly'
+import '@blockly/block-plus-minus'
+
 import Vue from 'vue'
 
 import defineOHBlocks from '@/assets/definitions/blockly/ohblocks'
+import defineOHBlocksTimers from '@/assets/definitions/blockly/ohblocks_timers'
+import defineOHBlocksEphemeris from '@/assets/definitions/blockly/ohblocks_ephemeris'
+import defineOHBlocksAudio from '@/assets/definitions/blockly/ohblocks_audio'
+import defineOHBlocksBusEvents from '@/assets/definitions/blockly/ohblocks_busevents'
+import defineOHBlocksLogging from '@/assets/definitions/blockly/ohblocks_logging'
 
-Vue.config.ignoredElements = ['field', 'block', 'category', 'xml', 'mutation', 'value', 'sep']
+Vue.config.ignoredElements = [
+  'field',
+  'block',
+  'category',
+  'xml',
+  'mutation',
+  'value',
+  'sep'
+]
 
 export default {
   props: ['blocks'],
   data () {
     return {
       workspace: null,
-      loading: true
+      sinks: [],
+      voices: [],
+      scripts: [],
+      rules: [],
+      addons: [],
```

Comment:
What is this for?

Reply:
This are used for the providing the options in the dropdowns.
Note that scripts is not used yet because it is one of the block-groups that I haven't worked on but is coming soon (see the thread post of me and Rich about bringing this up), so i would leave that in for now.
Regarding the addons I am not really sure why "bigbased" had put that in - I currently cannot see a usage for that so we might be able to remove that but it would mean I have to do a full regression test on all blocks to make sure it has no side effect taking it out.